runtime/tests/decimal_conv.c

```diff
@@ -68,7 +68,38 @@ void testIntToDec() {
     validateIntToDec(INT64_MIN + 1, "-9223372036854775807");
 }
 
+void validateFloatToDec(double val, const char *decStr, PanicCode code) {
+    validate(_bal_decimal_from_float(val), decStr, code);
+}
+
+void testFloatToDec() {
+    validateFloatToDec(0.0, "0", 0);
+    validateFloatToDec(1.0, "1", 0);
+    validateFloatToDec(1.00000, "1", 0);
+    validateFloatToDec(1234567891.0, "1234567891", 0);
+    validateFloatToDec(1234567890123456.0, "1234567890123456", 0);
+    validateFloatToDec(1234567890123456.1, "1234567890123456", 0);
+    validateFloatToDec(1234567890123456.2, "1234567890123456.3", 0);
```

Comment:
Why is it correct to have the `.3` here?

Reply:
Basically, this is the string output we get from `dtoa` function. But this does not violate the round-trip behavior. I have changed the tests to check round-trip.